src/backend/columnar/columnar_customscan.c

```diff
@@ -524,13 +637,43 @@ CreateColumnarScanPath(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte)
 	/* columnar scans are not parallel-aware, but they are parallel-safe */
 	path->parallel_safe = rel->consider_parallel;
 
+	path->param_info = get_baserel_parampathinfo(root, rel,
+												 requiredRelids);
+
 	/*
-	 * We don't support pushing join clauses into the quals of a seqscan, but
-	 * it could still have required parameterization due to LATERAL refs in
-	 * its tlist.
+	 * Now, baserestrictinfo contains the clauses referencing only this rel,
+	 * and ppi_clauses (if present) represents the join clauses that reference
+	 * this rel and rels contained in requiredRelids (after accounting for
+	 * ECs). Combine the two lists of clauses, extracting the actual clause
+	 * from the rinfo, and filtering out pseudoconstants and SAOPs.
 	 */
-	path->param_info = get_baserel_parampathinfo(root, rel,
-												 rel->lateral_relids);
+	List *allClauses = rel->baserestrictinfo;
+	if (path->param_info != NULL)
+	{
+		allClauses = list_concat(allClauses, path->param_info->ppi_clauses);
+	}
+
+	List *pushClauses = NIL;
+	ListCell *lc;
+	foreach (lc, allClauses)
+	{
+		RestrictInfo *rinfo = (RestrictInfo *) lfirst(lc);
```

Comment:
Nice, I like that. I am going to hold off on this slightly until we have a little more clarity on how columnar might land on FSPG. I don't want to add too many citus dependencies (though this would probably be fine because we could easily copy the nice macro).